packages/flutter_hooks/lib/src/framework.dart

```diff
@@ -221,13 +221,32 @@ abstract class HookState<R, T extends Hook<R>> with Diagnosticable {
   T get hook => _hook!;
   T? _hook;
 
+  VoidCallback? _onDidBuild;
+
+  /// Callback for when [HookWidget] build is finished
+  VoidCallback? get onDidBuild => _onDidBuild;
+
+  /// Register a callback for when [HookWidget] build is finished
+  /// Pass null to unregister the callback
+  set onDidBuild(VoidCallback? callback) {
+    _onDidBuild = callback;
+    if (callback != null) {
+      _element?.addHookForDidBuild(this);
```

Comment:
We need to check that the previous callback was null too before making this call.

Otherwise `onDidBuild = x; onDidBUild = x;` will add `this` twice to the list

Reply:
I'd also do `_element!.`

And possibly start the setter with an `assert(_element != null, 'Cannot update onDidBuild inside the constructor of a HookState')`

Reply:
Done, I understand now the `List` instead of `Set` ^^